### PR TITLE
feat: support decorator proposal

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit adds support for TC39's [decorator proposal](https://github.com/tc39/proposal-decorators).

A new "EmitOptions::use_ts_decorators" field has been added that defaults to false. If users wish to use
the TS decorators instead of decorator proposal, they should pass `true` for that field.

### CODE REVIEW

File: src/parsing.rs [hunk 1]
```diff
@@ -276,8 +276,8 @@ pub fn get_syntax(media_type: MediaType) -> Syntax {
       allow_return_outside_function: true,
       allow_super_outside_method: true,
       auto_accessors: true,
-      decorators: false,
-      decorators_before_export: false,
+      decorators: true,
+      decorators_before_export: true,
```
Comment (bartlomieju) [inline src/parsing.rs:280]: Two things:
- I'm not sure if this should be enabled by default, but adding a new argument to `get_syntax` is gonna be painful
- `decorators_before_exports` needs to be verified - I think it's a Babel transform that we don't necessarily want to enable

File: src/transpiling/mod.rs [hunk 2]
```diff
@@ -290,11 +292,15 @@ pub fn fold_program(
...
-    }),
+    Optional::new(
+      proposal::decorators::decorators(proposal::decorators::Config {
+        legacy: true,
+        emit_metadata: options.emit_metadata,
+  
+        use_define_for_class_fields: true,
+      }),
+      !options.use_decorator_proposal,
+    ),
```
Comment (bartlomieju) [inline src/transpiling/mod.rs:303]: Backwards compat - the TS experiemental decorators transform is only applied if `use_decorator_propsal` hasn't been configured

File: src/parsing.rs (hunk 1 above)
Comment (bartlomieju) [reply src/parsing.rs:280]: I did search through TC39 proposal repo but couldn't find an obvious answer. @nicolo-ribaudo any chance you could chime in here and suggest the proper default?

Comment (nicolo-ribaudo) [reply src/parsing.rs:280]: The proposal now settled to decorators _after_ export, so this should default to `false` (and ideally not be configurable at all by users). We are going to remove this option in the next Babel major btw.

Comment (bartlomieju) [reply src/parsing.rs:280]: Lovely, thanks!

**Review by dsherret — APPROVED** [review]
LGTM!

**Pull request merged**